Spigot-API-Patches/0082-Basic-PlayerProfile-interface.patch

```diff
@@ -0,0 +1,224 @@
+ import org.bukkit.inventory.ItemFactory;
+ import org.bukkit.inventory.meta.ItemMeta;
+ 
++import javax.annotation.Nullable;
++
+ /**
+  * Represents the Bukkit core, for version and Server singleton handling
+  */
+@@ -1216,6 +1218,34 @@ public final class Bukkit {
+     public static boolean suggestPlayerNamesWhenNullTabCompletions() {
+         return server.suggestPlayerNamesWhenNullTabCompletions();
+     }
++
++    /**
++     * Creates a PlayerProfile for the specified name/uuid
++     * @param uuid UUID to create profile for
++     * @return A PlayerProfile object
++     */
++    public static com.destroystokyo.paper.profile.PlayerProfile createProfile(UUID uuid) {
++        return server.createProfile(uuid);
++    }
++
++    /**
++     * Creates a PlayerProfile for the specified name/uuid
++     * @param name Name to create profile for
++     * @return A PlayerProfile object
++     */
++    public static com.destroystokyo.paper.profile.PlayerProfile createProfile(String name) {
++        return server.createProfile(name);
++    }
++
++    /**
++     * Creates a PlayerProfile for the specified name/uuid
++     * @param uuid UUID to create profile for
++     * @param name Name to create profile for
++     * @return A PlayerProfile object
++     */
++    public static com.destroystokyo.paper.profile.PlayerProfile createProfile(UUID uuid, String name) {
++        return server.createProfile(uuid, name);
++    }
+     // Paper end
+ 
+     public static Server.Spigot spigot()
+diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
+index 87ab9d2b..57eb0d90 100644
+--- a/src/main/java/org/bukkit/Server.java
++++ b/src/main/java/org/bukkit/Server.java
+@@ -45,6 +45,8 @@ import org.bukkit.generator.ChunkGenerator;
+ import org.bukkit.inventory.ItemFactory;
+ import org.bukkit.inventory.meta.ItemMeta;
+ 
++import javax.annotation.Nullable;
++
+ /**
+  * Represents a server implementation.
+  */
+@@ -1041,5 +1043,27 @@ public interface Server extends PluginMessageRecipient {
+      * @return true if player names should be suggested
+      */
+     boolean suggestPlayerNamesWhenNullTabCompletions();
++
++    /**
++     * Creates a PlayerProfile for the specified name/uuid
++     * @param uuid UUID to create profile for
++     * @return A PlayerProfile object
++     */
++    com.destroystokyo.paper.profile.PlayerProfile createProfile(UUID uuid);
++
++    /**
++     * Creates a PlayerProfile for the specified name/uuid
++     * @param name Name to create profile for
++     * @return A PlayerProfile object
++     */
++    com.destroystokyo.paper.profile.PlayerProfile createProfile(String name);
++
++    /**
++     * Creates a PlayerProfile for the specified name/uuid
++     * @param uuid UUID to create profile for
++     * @param name Name to create profile for
++     * @return A PlayerProfile object
++     */
++    com.destroystokyo.paper.profile.PlayerProfile createProfile(UUID uuid, String name);
```

Comment:
Maybe this should throw an exception if neither the uuid or the name are provided?

Reply:
Performed null check in CraftGameProfile instead.

Reply:
Actually, Mojangs own check handled that...

Reply:
I'm just thinking the documentation for that method should have a `throws` comment warning about this exception. The actual check can be done by the impl.